Update appearance of validation admin tables

delawski opened the pull request:
## Summary

This PR updates the look and feel of the AMP validation-related tables in the admin area.

As part of this work, the `amp-validation-error-taxonomy` stylesheet has been converted to SCSS and partially broken apart.

Also, some of the icons have been replaced with new ones ("invalid"/"kept" icon, "valid" icon). Additionally, a new icon has been introduced for the "removed" status. The only icon that has not been updated is the "warning" icon (a yellow circle with an exclamation mark inside). It's because we don't have a new version for that icon right now.

Note that the "Sources" icons have not been updated, too.

**Validated URLs**
<img width="1496" alt="Screenshot 2021-04-13 at 16 02 15" src="https://user-images.githubusercontent.com/478735/114565548-a6496f00-9c71-11eb-8118-267041320464.png">

**Validated URL**
<img width="1481" alt="Screenshot 2021-04-13 at 15 58 14" src="https://user-images.githubusercontent.com/478735/114564922-15729380-9c71-11eb-9ac9-99df7705cbb0.png">

**Validated URL - error details revealed**
<img width="1478" alt="Screenshot 2021-04-13 at 15 59 07" src="https://user-images.githubusercontent.com/478735/114565038-320ecb80-9c71-11eb-804a-439b0a31442a.png">

**Stylesheets**
<img width="1478" alt="Screenshot 2021-04-13 at 15 59 32" src="https://user-images.githubusercontent.com/478735/114565105-405ce780-9c71-11eb-9eab-517c500c15d8.png">

**Error Index**
<img width="1484" alt="Screenshot 2021-04-13 at 16 01 16" src="https://user-images.githubusercontent.com/478735/114565411-85811980-9c71-11eb-9b4f-30ecac215645.png">

**Validation Error**
<img width="1489" alt="Screenshot 2021-04-13 at 16 00 39" src="https://user-images.githubusercontent.com/478735/114565304-6e422c00-9c71-11eb-9183-74e33abcc46e.png">

<!-- Please reference the issue(s) this PR fixes, if one exists. For significant changes, opening an issue first for discussion is usually preferable. -->
Fixes #5995 

## Checklist

- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

delawski committed:
Use SCSS for error taxonomy stylesheet and start breaking it apart

delawski committed:
Indicate new (unreviewed) errors with grey background and bold text

delawski committed:
Indicate kept errors with red left border

delawski committed:
Fix stylelint issues

delawski committed:
Use new icon for down-triangle (chevron)

delawski committed:
Update style of markup status dropdown in validated URL table

delawski committed:
Merge branch 'add/css-webpack-watch' into add/5995-validation-tables-css

delawski committed:
Update AMP status icons

delawski committed:
Update error details notice appearance

delawski committed:
Indicate markup status on error index (taxonomy) page

delawski committed:
Remove redundant blank line

westonruter committed:
Merge branch 'develop' of github.com:ampproject/amp-wp into add/5995-validation-tables-css

* 'develop' of github.com:ampproject/amp-wp: (25 commits)
  Update snapshots and fix tests
  Remove references to AMP 'page' since it may be post; harmonize strings
  Use boolean return type for isDevToolsEnabled selector
  Add missing name prop for PluginDocumentSettingPanel in AMPDocumentSettingPanel
  Show AMP Validation sidebar button when all issues have been reviewed
  Omit AMPToolbarButton when AMP is not enabled
  Make entry path regex match simpler
  Show AMP toggle when Dev Tools are disabled
  Improve JS test coverage
  Ignore block editor plugins from test coverage
  Ignore block editor plugins in unit test
  Add more unit tests
  Fix broken E2E test
  Add unit tests to AMPDocumentStatusNotification
  Use `isSecondary` button prop instead of deprecated `isDefault`
  Add unit tests to AMPToggle component
  Use maybeIsPostDirty flag in unit tests
  Update Jest snapshots
  Combine validation status components in common parent
  Show dirty post message in pre-publish and document sidebars
  ...

westonruter committed:
Fix phpcs warning for variable assigment alignment

westonruter committed:
Remove unnecessary optional chaining and since not available in Node 12

westonruter commented:
I think the Validated URL paths should remain blue links:

![image](https://user-images.githubusercontent.com/134745/114630504-9a5cad80-9c6f-11eb-9e6d-7ef9b9f48008.png)

westonruter commented:
All rows are currently bolded. Shouldn't only the new ones be bolded?

Before | After
-------|-------
![image](https://user-images.githubusercontent.com/134745/114632371-70a58580-9c73-11eb-9e73-b36292837687.png) | ![image](https://user-images.githubusercontent.com/134745/114632402-7dc27480-9c73-11eb-97eb-60f8baa210ac.png)

This makes it consistent with the errors in the AMP Validation sidebar:

![image](https://user-images.githubusercontent.com/134745/114632467-9b8fd980-9c73-11eb-8644-d4e6701dd62d.png)

westonruter commented:
I think the boldness may currently be reversed from what it needs to be. Look at Gmail, where the first and third conversations here are unread:

![image](https://user-images.githubusercontent.com/134745/114632733-21138980-9c74-11eb-977c-471166f69a07.png)

Unread messages have white background and bold text. However, at the moment reviewed validation errors have a white background with normal font weight, and unreviewed errors have gray backgrounds and bold text:

![image](https://user-images.githubusercontent.com/134745/114632877-5ddf8080-9c74-11eb-82da-ad10c43cb28d.png)

westonruter commented:
> I think the Validated URL paths should remain blue links:

(On second thought, maybe not. It looks worse, even though it is more normal for a list table.)

pierlon commented:
(Aside from the inverted read/unread rows mentioned earlier)

![image](https://user-images.githubusercontent.com/16200219/114634638-825a4d80-9cb2-11eb-8ba7-399f9d420dd9.png)

Should the `removed` and `valid` icons be using the same color? At first glance I wasn't sure if the `remove` and `valid` icons were to convey the same meaning. Maybe revert back to the `warning` icon here?

![image](https://user-images.githubusercontent.com/16200219/114634833-ec72f280-9cb2-11eb-9ffd-7dc7025464d2.png)

pierlon commented on assets/css/src/components/_wp-list-table-markup-status-icon.scss:
```diff
@@ -0,0 +1,28 @@
+		background-size: 18px auto;
+	}
+
+	.amp-validation-error-status { // SELECT tag
```
Why not simply add the `select` tag?

```suggestion
	select.amp-validation-error-status {
```

pierlon commented on assets/src/amp-validation/amp-validated-urls-index.js:
```diff
@@ -6,6 +6,6 @@ import domReady from '@wordpress/dom-ready';
+import setValidationErrorRowsClasses from './set-validation-error-rows-classes';
 
-domReady( setValidationErrorRowsSeenClass );
+domReady( setValidationErrorRowsClasses );
```
It seems this JS file is no longer needed, as you can't toggle the reviewed or error status from the Validated URLs index page.

delawski replied:
Good catch! The entire JS file is removed in d92ab08fa9d8a84945939164811ba1b3aade1d1d.

westonruter replied:
@pierlon Is this not actually needed per my last screenshot in https://github.com/ampproject/amp-wp/pull/6067#issuecomment-821633507?

pierlon replied:
Yes the function is still being called via the enqueued `amp-validation-detail-toggle.js`:
https://github.com/ampproject/amp-wp/blob/23f423363afc524c1d58fb23bf75d53b8b01e330/assets/src/amp-validation/amp-validation-detail-toggle.js#L84

pierlon commented on assets/src/amp-validation/amp-validation-detail-toggle.js:
```diff
@@ -81,5 +81,5 @@ domReady( () => {
 			} );
 		} );
 
-	setValidationErrorRowsSeenClass();
```
Seems this is no longer necessary as this function is only relevant to the AMP Validated URL screen.

delawski replied:
So after checking this one it turns out that it is used on the AMP Validation Error Index screen, too. What's more, the code also (incorrectly) interfered with the AMP Validated URLs screen (add  `new` class to every table row).

So, in 32afaf51f1e4bd1ec650241e4f05db85201395b6 I've updated the `setValidationErrorRowsSeenClass()` implementation so that the `new` class is toggled only if the hidden input is in the `column-status`. Since the hidden input is in the `.column-error_status` column on the AMP Validated URLs screen, it's not affected.

What's more, the `amp-validation-detail-toggle.js` is enqueued on every AMP admin page so I decided to keep it there and [remove](https://github.com/ampproject/amp-wp/pull/6067/commits/32afaf51f1e4bd1ec650241e4f05db85201395b6#diff-12c01fb0e2315d77441bb412562430cf3ecb4d5aeb493bf79440ca8640f78a54L32) it from the `amp-validated-url-post-edit-screen.js` instead.

jwold commented:
> I think the Validated URL paths should remain blue links:

Since the `Details` button shows up on hover I think it's ok, in this instance, to keep the links black. 

> All rows are currently bolded. Shouldn't only the new ones be bolded?

I went back and looked at Gmail, and for some reason I think I had it backwards mentally. We should update it to:
Unread: Bold and white
Read: Normal and gray

---

I found an error with the context column, where the gray background behind `<div>` extends the full width of the column, makes it look a bit strange: https://v.usetapes.com/xoccPHSS9v

---

> Should the removed and valid icons be using the same color? At first glance I wasn't sure if the remove and valid icons were to convey the same meaning. Maybe revert back to the warning icon here?

Good question! I'd like the icons to be consistent, so happy to. make a new one if it's needed; just not sure.

delawski committed:
Merge branch 'develop' into add/5995-validation-tables-css

delawski committed:
Apply suggestions from code review

Co-authored-by: Pierre Gordon <16200219+pierlon@users.noreply.github.com>

delawski committed:
Remove unused JS for toggling row classes on Validated URLs index

delawski committed:
Set initial class names on the errors rows once and refactor

delawski committed:
Be more specific in jsdoc param type definition

delawski committed:
Revers boldness of new-vs-reviewed error rows

delawski committed:
Prevent minor bottom border gap in error details cell

delawski committed:
Ensure SUMMARY marker is rendered correctly on Safari

delawski committed:
Fix stylelint error

delawski commented:
> I went back and looked at Gmail, and for some reason I think I had it backwards mentally. We should update it to:
Unread: Bold and white
Read: Normal and gray

@jwold @westonruter The boldness of the rows is now reversed to match Gmail style (14a4dbd):

<img width="463" alt="Screenshot 2021-04-16 at 18 33 39" src="https://user-images.githubusercontent.com/478735/115055833-51f8f600-9ee2-11eb-9762-bb2d329416fb.png">

> I found an error with the context column, where the gray background behind <div> extends the full width of the column, makes it look a bit strange: https://v.usetapes.com/xoccPHSS9v

Thanks for catching that up, @jwold. It should be fixed with e90f5a7.

---

The open question is the warning icon. Are we keeping the old one, or are we creating a new one (I guess this would be preferred so that the icon style is consistent)?

delawski committed:
Reverse boldness of new-vs-reviewed error rows in block editor

delawski commented:
I'd almost forgot about the block editor. In 9fa332c the boldness of the error rows has been reversed in the editor, too:

<img width="282" alt="Screenshot 2021-04-16 at 18 41 43" src="https://user-images.githubusercontent.com/478735/115056746-7e614200-9ee3-11eb-9cca-09e3453296af.png">

westonruter committed:
Merge branch 'develop' of github.com:ampproject/amp-wp into add/5995-validation-tables-css

* 'develop' of github.com:ampproject/amp-wp:
  Replace Codecov bash uploader with GHA action counterpart

westonruter commented:
> The open question is the warning icon. Are we keeping the old one, or are we creating a new one (I guess this would be preferred so that the icon style is consistent)?

The old one would remain as it is used in other places, namely admin bar. See screenshots in https://github.com/ampproject/amp-wp/pull/4986.

westonruter commented:
Validated URL screen looks good:

![image](https://user-images.githubusercontent.com/134745/115089377-f4a37b80-9ec6-11eb-91cc-adf0b272da76.png)

AMP validation sidebar in block editor looks good:

Reviewed hidden | Reviewed shown | All expanded
-----------------|------------------|--------------
![image](https://user-images.githubusercontent.com/134745/115089511-37655380-9ec7-11eb-907a-f517b2422a8e.png) | ![image](https://user-images.githubusercontent.com/134745/115089534-45b36f80-9ec7-11eb-943b-bf0f523d9a28.png) | ![image](https://user-images.githubusercontent.com/134745/115089578-59f76c80-9ec7-11eb-9ba8-e169c42b23cd.png)

Error index looks good and consistent:

![image](https://user-images.githubusercontent.com/134745/115089703-b78bb900-9ec7-11eb-89be-8a38036dfdbf.png)

There's just one case that doesn't look quite right, and that is the Validated URLs post list table. When a validated URL has one or more validation errors that are unreviewed, the entire URL should appear as unread (white background with bold text), but currently they all appear with the “read” state:

![image](https://user-images.githubusercontent.com/134745/115089867-20733100-9ec8-11eb-800f-42811d953a7d.png)

westonruter commented:
> The old one would remain as it is used in other places, namely admin bar. See screenshots in #4986.

Oh, and I can see that the valid icon is now showing up in the admin bar:

![image](https://user-images.githubusercontent.com/134745/115094444-5ae1cb80-9ed2-11eb-971d-d1309ff0c4f8.png)

Nevertheless, the old warning icon is remaining:

![image](https://user-images.githubusercontent.com/134745/115094501-a72d0b80-9ed2-11eb-99d4-22da32db3e5d.png)

So either they both need to be updated to use the new icons in this PR, or they should both remain what they were as seen in the screenshots in #4986.

westonruter reviewed (changes requested):
It's close! Requesting changes per https://github.com/ampproject/amp-wp/pull/6067#issuecomment-821633507 and https://github.com/ampproject/amp-wp/pull/6067#issuecomment-821724995.

delawski commented:
> There's just one case that doesn't look quite right, and that is the Validated URLs post list table. When a validated URL has one or more validation errors that are unreviewed, the entire URL should appear as unread (white background with bold text), but currently they all appear with the “read” state:

I've actually removed that behavior in one of my latest commits as I thought it's not intended. I'll revert that change 👍

> So either they both need to be updated to use the new icons in this PR, or they should both remain what they were as seen in the screenshots in #4986.

I'll get that sorted out shortly.

delawski committed:
Ensure rows with unreviewed errors are styled on AMP Validated URLs page

delawski committed:
Ensure expand arrow is positioned relative to stylesheet details button

delawski committed:
Convert `amp-icons` to SCSS and use alt icons in admin bar

delawski committed:
Use SCSS stylelint preset and fix SCSS lint issues

delawski commented:
> > There's just one case that doesn't look quite right, and that is the Validated URLs post list table. When a validated URL has one or more validation errors that are unreviewed, the entire URL should appear as unread (white background with bold text), but currently they all appear with the “read” state:

> I've actually removed that behavior in one of my latest commits as I thought it's not intended. I'll revert that change 👍

This has been addressed in bd64550.

---

> >So either they both need to be updated to use the new icons in this PR, or they should both remain what they were as seen in the screenshots in #4986.

> I'll get that sorted out shortly.

In 9d04eb4 the previous icons are used in the admin bar again:

<img width="90" alt="Screenshot 2021-04-19 at 19 02 48" src="https://user-images.githubusercontent.com/478735/115275430-12304980-a142-11eb-8675-6c7126d26d26.png">
<img width="96" alt="Screenshot 2021-04-19 at 19 03 04" src="https://user-images.githubusercontent.com/478735/115275434-13617680-a142-11eb-9bf9-cfa7198601ea.png">
<img width="84" alt="Screenshot 2021-04-19 at 19 03 14" src="https://user-images.githubusercontent.com/478735/115275436-13617680-a142-11eb-8c68-c48691c4bd75.png">

westonruter committed:
Preserve hover color for row-title

westonruter committed:
Remove obsolete Icon::get_color()

westonruter reviewed (approved):
Great work!

@pierlon Anything else from your perspective?

pierlon commented:
Experiencing a weird bug with the error details:


https://user-images.githubusercontent.com/16200219/115323461-b5e81c80-a177-11eb-9e05-2241bb0d5628.mp4

Notice how the arrows under the "Error" column are pointing down, while the ones under the "Context" column are pointing up. Clicking on any of the summaries under the "Context" column does not toggle the arrow, but doing so for the "Error" column does. Now the weird part is only after expanding the error details does the context arrow behave as expected.

pierlon commented on assets/css/src/amp-validation-error-taxonomy.scss:
```diff
@@ -59,13 +69,15 @@ details[open] .details-attributes__summary {
 	background-image: url("../images/down-triangle.svg");
-	background-size: cover;
 	background-position: center;
+	background-size: cover;
 	content: "";
+	height: 6px;
+	position: absolute;
+	right: 0;
+	top: 7px;
+	width: 12px;
```
This is now causing the arrow for stylesheet summary details to be misplaced:


https://user-images.githubusercontent.com/16200219/115323210-322e3000-a177-11eb-85f2-e42ee014f999.mp4

pierlon replied:
Maybe this arrow can be removed, since the summary marker should be enough of an indicator in this case.

delawski replied:
The original implementation removes the summary marker and uses the chevron instead. In 38d1104 this has been reverted:

<img width="347" alt="Screenshot 2021-04-20 at 11 45 14" src="https://user-images.githubusercontent.com/478735/115376525-0f2f6a80-a1cf-11eb-9bd7-4f97e1aeede4.png">

What's more, for some reason the `<strong>` element not always got the `source` class applied. This, in turn, caused a slight misalignment between icons in the "Sources" column. With 218eeef this is fixed:

| Before | After |
| --- | --- |
| <img width="138" alt="Screenshot 2021-04-20 at 11 54 59" src="https://user-images.githubusercontent.com/478735/115376992-7b11d300-a1cf-11eb-9996-2a5bfa9d25d6.png"> | <img width="143" alt="Screenshot 2021-04-20 at 11 55 08" src="https://user-images.githubusercontent.com/478735/115377011-806f1d80-a1cf-11eb-8588-a1d29e69c6e3.png"> |

delawski committed:
Keep DETAILS closed initially so that expand arrows stay in sync

delawski committed:
Position stylesheets expand arrow correctly and clean up CSS

delawski committed:
Extract error details toggle button to separate SCSS partial

delawski committed:
Extract details column to separate SCSS partial

delawski committed:
Remove legacy styles for single error detail summary

There is no reference to `.single-error-detail` in the plugin folder. The CSS code seems to be redundant.

delawski committed:
Ensure the `source` class is always used

delawski commented:
> Experiencing a weird bug with the error details:

@pierlon This was pretty odd.

It turns out that the `<details>` element in this particular location had the `open` property applied initially. Because of that, the toggles were out of sync. Once you clicked on the "Error" column, our JS kicked in and put them back in sync.

The simple solution was to remove the `open` property from the `<details>` element which has been done in a1c73d8

<img width="360" alt="Screenshot 2021-04-20 at 11 45 21" src="https://user-images.githubusercontent.com/478735/115376275-da231800-a1ce-11eb-97d6-ff59310bcc06.png">

delawski commented:
@westonruter While addressing @pierlon feedback I've done a bit of a clean-up around the CSS code I was touching. There should be no substantial changes in the compiled CSS files (aside from the `.single-error-detail` [removal](https://github.com/ampproject/amp-wp/pull/6067/commits/5cc9c1b39821313171f21902d8f2ccc9b8e2aa93) as it seems to not be present in the codebase anymore).

pierlon commented on assets/css/src/components/_error-details-toggle.scss:
```diff
@@ -0,0 +1,37 @@
+		}
+	}
+
+	&.is-open & {
```
The chevron for the "Context" column no longer expands:


https://user-images.githubusercontent.com/16200219/115483170-cc59ab00-a23f-11eb-9006-157b918705cd.mp4

```suggestion
	&.is-open {
```

delawski replied:
Thank you for catching that, @pierlon 🙏

pierlon committed:
Fix chevron for "Context" column not expanding

pierlon reviewed (approved):
Well done @delawski :ok_hand:.

pierlon commented:
https://github.com/ampproject/amp-wp/pull/6081 should fix the CI job failures here.

westonruter committed:
Merge branch 'develop' of github.com:ampproject/amp-wp into add/5995-validation-tables-css

* 'develop' of github.com:ampproject/amp-wp:
  Bump `codecov/codecov-action` to v1.4.1

westonruter reviewed (approved):
@pierlon Thanks a lot for the thorough review.

@delawski And thanks again for this great work. You're modernizing the CSS as you go which will make the reactification of these screens one step closer.
